src/mlpack/methods/kernel_svm/kernel_svm_function_impl.hpp

```diff
@@ -0,0 +1,226 @@
+    const arma::Row<size_t>& labels,
+    const double regularization,
+    const bool fitIntercept,
+    const size_t firstClasslabel,
+    const size_t secondClasslabel,
+    const size_t maxIter,
+    const double tol,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    firstClasslabel(firstClasslabel),
+    secondClasslabel(secondClasslabel),
+    kernel(kernel)
+{
+  intercept = 0;
+  Train(data, labels, maxIter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVMFunction<MatType, KernelType>::KernelSVMFunction(
+    const double regularization,
+    const bool fitIntercept,
+    const size_t firstClasslabel,
+    const size_t secondClasslabel,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    firstClasslabel(firstClasslabel),
+    secondClasslabel(secondClasslabel),
+    kernel(kernel)
+{
+  intercept = 0;
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVMFunction<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t maxIter,
+    const double tol)
+{
+  // Saving values of sample data to be used
+  // with kernel function.
+  trainingData = data;
+
+  // Changing labels values to 1, -1 values provided
+  // by user should 0 and 1.
+  trainCoefficients = arma::ones(1, data.n_cols);
+  trainCoefficients.cols(arma::find(labels == secondClasslabel)) += -2;
+
+  // Intializing variable to calculate alphas.
+  alpha = arma::zeros(1, data.n_cols);
+  size_t count = 0;
+
+  // Vector to store error value.
+  arma::vec E = arma::zeros(data.n_cols, 1);
+  // Eta value.
+  double eta = 0;
+
+  // Variable to store lower bound and higher bound.
+  double L = 0;
+  double H = 0;
+
+  // Storing kernel function values.
+  arma::mat K = arma::mat(data.n_cols, data.n_cols);
+
+  // Pre-compute the Kernel Matrix
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    for (size_t j = 0; j < data.n_cols; j++)
+    {
+      K(i, j) = kernel.Evaluate(data.col(i), data.col(j));
+    }
+  }
+
+  // Training starts from here.
+  while (count < maxIter)
+  {
+    size_t changedAlphas = 0;
+    for (size_t i = 0; i < data.n_cols; i++)
+    {
+      // Calculate Ei = f(x(i)) - y(i) using (2).
+      // E(i) = b + sum (X(i, :) * (repmat(alphas.*Y,1,n).*X)') - Y(i);
+      E(i) = intercept + arma::as_scalar(arma::sum(alpha
+             % trainCoefficients % K.col(i).t()))- trainCoefficients(i);
+
+      if ((trainCoefficients(i) * E(i) < -tol && alpha(i) < regularization) ||
+        (trainCoefficients(i) * E(i) > tol && alpha(i) > 0))
+      {
+        // In practice, there are many ways one can use to select
+        // the i and j. In this simplified code, we select them randomly.
+        size_t j = rand() % data.n_cols;
+        while (j == i)
+        {
+          j = rand() % data.n_cols;
+        }
+        assert(j < data.n_cols && j >= 0);
```

Comment:
I guess, with the while loop above, that should be always true?

Reply:
No in case if `size_t j = rand() % data.n_cols;` generates a value which is equal to `i` then this loop will run and change the value otherwise no need of this loop.